### DISCUSSION TITLE
Ability to configure anisotropy and texture filtering

### DISCUSSION THREAD

**Discussion by enslaved2die** [google/model-viewer#3659] [Q&A]
I really want to see a option to set the anisotropic filtering level for textures.
I would  be really cool to use the model viewer e.g. for previewing different fabrics, brick or other materials, often you want to get a little bit closer and looking on these materials on a really sharp angle and than the texture falls apart due to simple bilinear filtering or a quite low anisotropic filtering level.

**Comment by cdata** [comment]
@enslaved2die very interesting suggestion. [Three.js' has configuration for texture anisotropy](https://threejs.org/docs/#api/en/textures/Texture.anisotropy), as well as texture filter configuration. I wonder how we could expose this in a generally useful way 🤔

cc @elalish

**Comment by elalish** [comment]
@enslaved2die Thanks, would it be possible for you to share a model or two with us to demonstrate the problem? I would want to use these to verify a solution. You can also include screenshots from https://modelviewer.dev/examples/tester.html to point us to the precise error.

**Comment by enslaved2die** [comment]
Of course.
Here are two fabric examples.
No compression at all to show the ground truth. (This is also why its on my GDrive, sry for that)
https://drive.google.com/drive/folders/1g9pTIEtOAVf5otZFHkYGsE0F3r4d2fkT?usp=sharing

Here you can see the exact issue aswell:
![Bildschirmfoto 2020-03-03 um 16 39 53](https://user-images.githubusercontent.com/25615199/75792545-6aee2880-5d6e-11ea-99d0-8533b97eed83.png)
The textures are really blurry on deep angles. I think a option to set the Anisotropic Filtering level could help here.

Thanks in advance :)

**Comment by ghost** [comment]
I am also having this issue, where the texture gets blurry on a low angle. In another model viewing application I can set the Texture Filtering to Linear and that solves the issue. Is there a way to achieve this in the model-viewer?

![image](https://user-images.githubusercontent.com/7875817/144909811-1837f825-b714-46bb-9df5-a463b858a7ce.png)

**Comment by elalish** [comment]
@braedenf Linear/Nearest is a different thing than the anisotropy discussion above. Linear/nearest is actually set in the glTF (though it's nearly always an error to do so, as the defaults give the best results). Try removing any filters under `Sampler` and see if that helps. The glTF VSCode extension makes this fairly easy.

**Comment by ghost** [comment]
Ah now I understand, thanks for you help.
